src/content/get-started/data-and-json.md

```diff
@@ -0,0 +1,466 @@
+---
+title: Data and JSON
+short-title: Data and JSON
+description: >-
+  Learn about JSON deserialization in Dart, and how to use `dart:convert`,
+  `jsonDecode`, and pattern matching to work with JSON data from the Wikipedia
+  API.
+prevpage:
+  url: /get-started/command-runner-polish
+  title: Command runner polish
+nextpage:
+  url: /get-started/testing
+  title: Testing
+---
+
+{% include 'fwe-wip-warning.md' %}
+
+In this chapter, you'll learn how to work with JSON (JavaScript Object Notation)
+data in Dart. JSON is a common format for data exchange on the web, and you'll
+often encounter it when working with APIs. You'll learn how to convert JSON data
+into Dart objects, making it easier to work with in your application. You'll use
+the `dart:convert` library, the `jsonDecode` function, and pattern matching.
+
+:::secondary What you'll learn
+
+*  Deserialize JSON data into Dart objects.
+*  Use the `dart:convert` library to work with JSON.
+*  Use the `jsonDecode` function to parse JSON strings.
+*  Use pattern matching to extract data from JSON objects.
+*  Create Dart classes to represent JSON data structures.
+
+:::
+
+## Prerequisites
+
+Before you begin this chapter, ensure you:
+
+* Have completed Chapter 8 and have a working Dart development environment with
+  the `dartpedia` project.
+* Understand basic Dart syntax, including classes and data types.
+
+## Tasks
+
+In this chapter, you'll create Dart classes to represent the JSON data returned
+by the Wikipedia API. This will allow you to easily access and use the data in
+your application.
+
+### Task 1: Create the Wikipedia package
+
+First, you'll create a new Dart package to house the data models.
+
+1.  Navigate to the root directory of your project (`/dartpedia`).
+2.  Run the following command in your terminal:
+
+    ```bash
+    dart create wikipedia
+    ```
+
+    This command creates a new directory named `wikipedia` with the basic
+    structure of a Dart package. You should now see a new folder
+    `wikipedia` in your project root, alongside `cli` and `command_runner`.
+
+### Task 2: Configure a Dart workspace
+
+Dart workspaces allow you to manage multiple related packages within a single
+project, simplifying dependency management and local development. Now that
+you're adding your 3rd package, its a good time to configure your project to use
+a Dart workspace.
+
+1.  **Create the root `pubspec.yaml` file.**
+    
+    Navigate to the root directory of your project (`/dartpedia`) and create a
+    new file named `pubspec.yaml` with the following content:
+
+    ```yaml
+    name: _
+    publish_to: none
+    environment:
+        sdk: ^3.8.1 # IMPORTANT: Adjust this to match your Dart SDK version or a compatible range (e.g., ^3.4.4 if you're on the latest stable)
```

Comment:
```suggestion
        sdk: ^3.8.1 # IMPORTANT: Adjust this to match your Dart SDK version or a compatible range
```

Reply:
We're on 3.7 as of today (3.8 next week), and pub workspaces requires version 3.6 or higher.  For longevity, just leave the 'latest' comment out.